SCons/Tool/MSCommon/vc.py

```diff
@@ -192,44 +347,100 @@ def get_msvc_version_numeric(msvc_version):
     """
     return ''.join([x for  x in msvc_version if x in string_digits + '.'])
 
-def get_host_target(env):
-    host_platform = env.get('HOST_ARCH')
-    debug("HOST_ARCH:%s", str(host_platform))
-    if not host_platform:
-        host_platform = platform.machine()
+def get_host_platform(host_platform):
+
+    host_platform = host_platform.lower()
 
     # Solaris returns i86pc for both 32 and 64 bit architectures
-    if host_platform == "i86pc":
+    if host_platform == 'i86pc':
         if platform.architecture()[0] == "64bit":
             host_platform = "amd64"
         else:
             host_platform = "x86"
 
```

Comment:
Need to retain debug() output from original code.

Reply:
The debug output sequence is identical to the existing code.  It is a bit more indirect  compared to the existing code as the the Solaris fix was moved into a function that is called from the two new functions.

Current master snippet:
```
def get_host_target(env):

    host_platform = env.get('HOST_ARCH')
    debug("HOST_ARCH:%s", str(host_platform))
    if not host_platform:
        host_platform = platform.machine()

    # Solaris returns i86pc for both 32 and 64 bit architectures
    if host_platform == "i86pc":
        if platform.architecture()[0] == "64bit":
            host_platform = "amd64"
        else:
            host_platform = "x86"

    # Retain user requested TARGET_ARCH
    req_target_platform = env.get('TARGET_ARCH')
    debug("TARGET_ARCH:%s", str(req_target_platform))
    if req_target_platform:
        # If user requested a specific platform then only try that one.
        target_platform = req_target_platform
    else:
        target_platform = host_platform

   ....
```
Current proposed snippet (Annotated):
```
def get_host_target(env, msvc_version, all_host_targets=False):

   ...

    host_arch = env.get('HOST_ARCH') if env else None
    debug("HOST_ARCH:%s", str(host_arch))

    # JCB NOTE: both of the following may use Solaris fix
    if host_arch:
        host_platform = get_host_platform(host_arch)
    else:
        host_platform = get_native_host_platform()

    target_arch = env.get('TARGET_ARCH') if env else None
    debug("TARGET_ARCH:%s", str(target_arch))

```

In the existing code and the proposed code, the HOST_ARCH is written to the debug output before the potential Solaris fix followed by the TARGET_ARCH.

The returned host platform is written to the debug output following the call to get_host_target in two places in both the existing code and the proposed code.

The debug output sequence is the same for the existing code and the proposed code.

I'm happy to re-work the debug output if you think it is necessary.